### ISSUE TITLE
Xournal++ crashes after input with a Gaomon Tablet

### ISSUE THREAD

**Issue by tildezero** [xournalpp/xournalpp#1788]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Win 10 
 - Version: 1.0.17
 - Installation method: ZIP File from official GitHub release

**Describe the bug**
When opening X++ on a new note, then trying to make a stroke w/ tablet, the program crashes as soon as the pen touches tablet

**To Reproduce**
1. Open Xournal++
2. attempt to write something with tablet and pen
3. program crashes

**Expected behavior**
The pen input should have made a stroke on the paper

**Additional context**
Experimental input is off, pen shows up correctly as input device\

Sometimes, no input devices show up also

**Comment by lehmanju** [comment]
@suhasvoolla could you enable the experimental input system and try again? the old input system isn't really maintained anymore and up for removal in a future release, try to set the correct device class if drawing doesn't work out of the box

**Comment by suhasvoolla** [comment]
I set the device class, and turned on new input, but still crashes

**Comment by suhasvoolla** [comment]
I retested, and now it isn't crashing, but it is not registering a lot of the strokes, and if it does, the input lag is really bad, and the buttons are not working on the pen

**Comment by Febbe** [comment]
Did you configured your stylus buttons in the settings, often they are not configured correctly by default :/?

**Comment by suhasvoolla** [comment]
yes, i configured b1 as eraser, and the other button does not do anything and I do not have an eraser

**Comment by mgavin** [comment]
This is likely unrelated to the previous issue, but Xournalpp crashes after upgrading the latest firmware for GAOMON for the GAOMON drawing tablet/screen (device PD2200). It may suggest that the issue is with the firmware itself, but I'm not sure what to make of the crash log (or how to get a more detailed crash log) when doing some various tests.

What's interesting is that when disabling everything, events for BUTTON[1-3] crash the program (b/c gdk?), but when the buttons (specifically the ones you can set, 2-3) are assigned to anything other than Mouse Click Middle / Mouse Right-click Button, it doesn't crash the application.

Basically I'm hitting this state:

```
===> GDK_MOTION_NOTIFY 0000000000170e3e @ 518838843ms (1034,302.9) (3963,531.9)  DefWindowProcW => 0
WT_PACKET 0000000000120da8 0x6a 0x3e8 106 00000000000003e8
  WM_NCHITTEST 0000000000170e3e 0 0x2120f7a DefWindowProcW => 1gdk_input_other_event: window=0000000000170e3e +1033+301
WINTAB motion: 1034.3,302.507

  ===> GDK_MOTION_NOTIFY 0000000000170e3e @ 518838843ms (1034,302.5) (3963,531.5)  DefWindowProcW => 0
WT_PACKET 0000000000120da8 0x6b 0x3e8 107 00000000000003e8
  WM_NCHITTEST 0000000000170e3e 0 0x2120f7a DefWindowProcW => 1gdk_input_other_event: window=0000000000170e3e +1033+301
WINTAB motion: 1033.95,302.145

  ===> GDK_MOTION_NOTIFY 0000000000170e3e @ 518838843ms (1034,302.1) (3963,531.1)  DefWindowProcW => 0
WT_PACKET 0000000000120da8 0x6c 0x3e8 108 00000000000003e8
  WM_NCHITTEST 0000000000170e3e 0 0x2120f7a DefWindowProcW => 1gdk_input_other_event: window=0000000000170e3e +1033+301
WINTAB button press:1 1033.51,301.736

  ===> GDK_BUTTON_PRESS 0000000000170e3e @ 518838843ms 1 (1034,301.7) (3963,530.7) BUTTON1
(xournalpp_bin.exe:21032): Gdk-CRITICAL **: 15:09:48.212: gdk_device_get_source: assertion 'GDK_IS_DEVICE (device)' failed
```

for BUTTON1, BUTTON2, BUTTON3.

Enabling / disabling "experimental input system" shows no difference.
Administrative mode on the Gaomon software makes no difference.
Enabling / disabling the pen input itself makes no difference... so... I'm at a loss... 

is this a gtk thing? ... and if so, what? what was different before? I'd have to go back and reinstall the previous gaomon software... ugh

**Comment by mgavin** [comment]
follow up to above,

I switched back to the old driver, and xournalpp is printing debug events with events prefixed with WM_, and GDK_BUTTON_PRESS events... so... sigh

**Comment by JonOfUs** [comment]
I can confirm this bug/crash. I cannot get Xournal (v 1.0.20, Windows) working with my Gaomon S620 Tablet with the newest firmware (16.0.0.20). I tried it directly after a clean install (Xournal++ was never installed before).
Every time I start trying to draw anything, Xournal++ closes instantaneously without any error message - and without asking to save the current note.
I tried activating the new experimental input system as well as deactivating the pressure sensitivity, but the crash persistently happened all the time.
Drawing with my mouse is no problem, it even works fine while the Gaomon Tablet is connected to my PC.

I don't know Xournal++ very well, just started using it, so I don't know where logs are located within the program. I could send those if someone could tell me where to find them.

**Comment by rolandlo** [comment]
This seems to have been [fixed](https://gitlab.gnome.org/GNOME/gtk/-/commit/af747315d79331b33b3967f39103d73b471ec658) in GTK 3.24.31, which has been released on Dec 20 and will soon land in Xournal++ nightlies through the Azure Pipelines.
